tests/acceptance/features/bootstrap/RoleAssignmentContext.php

```diff
@@ -0,0 +1,246 @@
+	 * @throws GuzzleException
+	 * @throws Exception
+	 */
+	public function sendRequestToAssignRoleToUser(string $user, string $userId, string $roleId): void {
+		$fullUrl = $this->baseUrl . $this->setttingsUrl . "assignments-add";
+		$body = json_encode(["account_uuid" => $userId, "role_id" => $roleId], JSON_THROW_ON_ERROR);
+
+		$this->featureContext->setResponse($this->spacesContext->sendPostRequestToUrl($fullUrl, $user, $this->featureContext->getPasswordForUser($user), $body));
+	}
+
+	/**
+	 * @param string $user
+	 * @param string $userId
+	 *
+	 * @return void
+	 *
+	 * @throws GuzzleException
+	 * @throws Exception
+	 */
+	public function sendRequestAssignmentsList(string $user, string $userId): void {
+		$fullUrl = $this->baseUrl . $this->setttingsUrl . "assignments-list";
+		$body = json_encode(["account_uuid" => $userId], JSON_THROW_ON_ERROR);
+
+		$this->featureContext->setResponse($this->spacesContext->sendPostRequestToUrl($fullUrl, $user, $this->featureContext->getPasswordForUser($user), $body));
+	}
+
+	/**
+	 * @When /^the administrator has given "([^"]*)" the role "([^"]*)" using the settings api$/
+	 *
+	 * @param string $user
+	 * @param string $role
+	 *
+	 * @return void
+	 *
+	 * @throws GuzzleException
+	 * @throws Exception
+	 */
+	public function theAdministratorHasGivenUserTheRole(string $user, string $role): void {
+		$admin = $this->featureContext->getAdminUserName();
+		$roleId = $this->userGetRoleIdByRoleName($admin, $role);
+		$userId = $this->featureContext->getAttributeOfCreatedUser($user, 'id');
+		$this->setRoleToUser($admin, $userId, $roleId);
+	}
+
+	/**
+	 * @param string $user
+	 * @param string $role
+	 *
+	 * @return string
+	 */
+	public function userGetRoleIdByRoleName($user, $role): string {
+		$this->getAllExistingRoles($user);
+
+		if ($this->featureContext->getResponse()) {
+			$rawBody =  $this->featureContext->getResponse()->getBody()->getContents();
+			if (isset(\json_decode($rawBody, true, 512, JSON_THROW_ON_ERROR)["bundles"])) {
+				$bundles = \json_decode($rawBody, true, 512, JSON_THROW_ON_ERROR)["bundles"];
+			}
+		}
+
+		$roleToAssign = "";
+		foreach ($bundles as $value) {
+			// find the selected role
+			if ($value["displayName"] === $role) {
+				$roleToAssign = $value;
+				break;
+			}
+		}
+		Assert::assertNotEmpty($roleToAssign, "The selected role $role could not be found");
+		return $roleToAssign["id"];
+	}
+
+	/**
+	 * @param string $user
+	 * @param string $userId
+	 * @param string $roleId
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function setRoleToUser($user, $userId, $roleId): void {
+		$this->sendRequestToAssignRoleToUser($user, $userId, $roleId);
+
+		if ($this->featureContext->getResponse()) {
+			$rawBody = $this->featureContext->getResponse()->getBody()->getContents();
+			if (isset(\json_decode($rawBody, true, 512, JSON_THROW_ON_ERROR)["assignment"])) {
+				$assignment = \json_decode($rawBody, true, 512, JSON_THROW_ON_ERROR)["assignment"];
+			}
```

Comment:
same here, if there is no "assignment" then the code further down will fail with errors that are hard to understand